instance-scheduler/environments.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"encoding/json"
+	//"errors"
+	//"log"
+	"strings"
+	"fmt"
+    "net/http"
+    "net/url"
+    "io/ioutil"
+
+)
+
+// Additional functions that parse json data from the environments directory obtail the full list of in-scope non-prod environments.
+
+// GitHubFile represents a single file in a GitHub directory as returned by the GitHub API
+type GitHubFile struct {
+    Name string `json:"name"`
+    Path string `json:"path"`
+    Type string `json:"type"` // "file" or "dir"
+}
+
+// JSONFileContent represents the structure of each JSON file's content
+type JSONFileContent map[string]interface{}
+
+// FetchJSON fetches the JSON content from a given URL
+func FetchJSON(rawURL string) (JSONFileContent, error) {
+    resp, err := http.Get(rawURL)
+    if err != nil {
+        return nil, fmt.Errorf("failed to fetch JSON content: %w", err)
+    }
+    defer resp.Body.Close()
+
+    if resp.StatusCode != http.StatusOK {
+        return nil, fmt.Errorf("non-200 status code: %d", resp.StatusCode)
+    }
+
+    body, err := ioutil.ReadAll(resp.Body)
```

Comment:
can swap for the io package 🚀 

```suggestion
    body, err := io.ReadAll(resp.Body)
```